queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/shares.py

```diff
@@ -71,6 +71,73 @@ def update_share_structure(business: Business, share_structure: Dict) -> Optiona
     return err
 
 
+def update_share_structure_correction(business: Business, share_structure: Dict) -> Optional[List]:
+    """Manage the share structure for a business.
+
+    Assumption: The structure has already been validated, upon submission.
+
+    Other errors are recorded and will be managed out of band.
+    """
+    if not business or not share_structure:
+        # if nothing is passed in, we don't care and it's not an error
+        return None
+
+    err = []
+
+    # Delete the ones that are present in db but not in the json and create the ones in json but not in db.
+    if resolution_dates := share_structure.get('resolutionDates'):
+        # Two lists of dates in datetime format
+        business_dates = [item.resolution_date for item in business.resolutions]
+        parsed_dates = [parse(resolution_dt).date() for resolution_dt in resolution_dates]
+
+        # Dates in both db and json
+        inclusion_dates = [business.resolutions[index] for index, date in enumerate(business_dates)
+                           if date in parsed_dates]
+        if len(inclusion_dates) > 0:
+            business.resolutions = inclusion_dates
+
+        # Dates in json and not in db
+        exclusion_dates = [date for date in parsed_dates if date not in business_dates]
+
+        resolution_dates = exclusion_dates if len(exclusion_dates) > 0 else parsed_dates
+
+        for resolution_dt in resolution_dates:
+            try:
+                d = Resolution(
+                    resolution_date=resolution_dt,
+                    resolution_type=Resolution.ResolutionType.SPECIAL.value
+                )
+                business.resolutions.append(d)
+            except (ValueError, OverflowError):
+                err.append(
+                    {'error_code': 'FILER_INVALID_RESOLUTION_DATE',
+                     'error_message': f"Filer: invalid resolution date:'{resolution_dt}'"}
+                )
+
+    if share_classes := share_structure.get('shareClasses'):
+        try:
+            delete_existing_shares(business)
+        except:  # noqa:E722 pylint: disable=bare-except; catch all exceptions
+            err.append(
+                {'error_code': 'FILER_UNABLE_TO_DELETE_SHARES',
+                 'error_message': f"Filer: unable to delete shares for :'{business.identifier}'"}
+            )
+            # we're FUBAR, do not load the new shares
+            return err
+
+        try:
+            for share_class_info in share_classes:
+                share_class = create_share_class(share_class_info)
+                business.share_classes.append(share_class)
+        except KeyError:
+            err.append(
+                {'error_code': 'FILER_UNABLE_TO_SAVE_SHARES',
+                 'error_message': f"Filer: unable to save new shares for :'{business.identifier}'"}
+            )
```

Comment:
As per Lekshimi's previous comment, this needs to actually only remove the share classes that have been removed and add the new ones.  This code is removing all share classes associated with the business and adding all old and new share classes from scratch.  It's problematic as we will not be able to re-construct the versioning history accurately.

Reply:
Sure. I will implement this for `shareClass` in just in case this method is used later on.